cronjob_scripts/trigger-package-build.py

```diff
@@ -0,0 +1,217 @@
+    crates_to_check = random.sample(list(possible_crates), 4 * check_limit)
+    if os.environ.get("RECHECK"):
+        crates_to_check = ["cargo-quickinstall"] + crates_to_check
+
+    repo_url = get_repo_url()
+    branch = get_branch()
+    builds_scheduled = 0
+    for crate in crates_to_check:
+        print(f"Checking {crate} for {target_arch}")
+        version = get_current_version_if_unbuilt(
+            repo_url=repo_url, crate=crate, target_arch=target_arch
+        )
+        if not version:
+            continue
+
+        no_default_features = "false"
+        if crate == "gitoxide":
+            features = "max-pure"
+            no_default_features = "true"
+        elif crate == "sccache":
+            features = "vendored-openssl"
+        else:
+            features = ",".join(
+                [feat for feat in version["features"].keys() if "vendored" in feat]
+            )
+
+        workflow_run_input = {
+            "crate": crate,
+            "version": version["vers"],
+            "target_arch": target_arch,
+            "features": features,
+            "no_default_features": no_default_features,
+            "build_os": build_os,
+            "branch": branch,
+        }
+        print(f"Attempting to build {crate} {version['vers']} for {target_arch}")
+        subprocess.run(
+            ["gh", "workflow", "run", "build-package.yml", "--json", f"--ref={branch}"],
+            input=json.dumps(workflow_run_input).encode(),
+            check=True,
+        )
+        builds_scheduled += 1
+        if builds_scheduled >= check_limit:
+            print("Scheduled enough builds, exiting")
+            break
+        time.sleep(30)
+
+
+def get_build_os(target_arch: str) -> str:
+    try:
+        return TARGET_ARCH_TO_BUILD_OS[target_arch]
+    except KeyError:
+        raise ValueError(f"Unrecognised target arch: {target_arch}")
+
+
+def get_target_architectures() -> list[str]:
+    target_arch = os.environ.get("TARGET_ARCH", None)
+    if target_arch in TARGET_ARCH_TO_BUILD_OS:
+        return [target_arch]
+
+    if target_arch == "all":
+        return list(TARGET_ARCH_TO_BUILD_OS.keys())
+
+    try:
+        rustc_version_output = subprocess.run(
+            ["rustc", "--version", "--verbose"], capture_output=True, text=True
+        )
+        current_arch = [
+            line.split("host: ")[1]
+            for line in rustc_version_output.stdout.splitlines()
+            if line.startswith("host: ")
+        ][0]
+        return [current_arch]
+    except:
+        raise ValueError(f"rustc did not tell us its host: {rustc_version_output}")
+
+
+def get_popular_crates():
+    with open(
+        Path(__file__).resolve().parent.parent.joinpath("popular-crates.txt"), "r"
+    ) as file:
+        return [
+            line.strip()
+            for line in file
+            if line.strip() != "" and not line.startswith("#")
+        ]
```

Comment:
Something that we could consider using here:

https://github.com/cargo-bins/cargo-quickinstall/issues/268#issuecomment-2329308074

Reply:
I will add a comment. In practice, that would probably still be put into a text file by some daily cronjob, so the shape of this function probably won't change that much? I haven't really thought about this enough.

Reply:
crates-io dump is updated daily, and I think we could use `actions/cache` to cache it?

Reply:
yeah, possibly, assuming it's not too big. Something for a later PR though.

Reply:
Think it's simple enough, was polars I was able to do it around 10 lines, and it's quite efficient (in the linked issue)